server/src/main/java/com/flipkart/varadhi/web/AuthorizationHandlerBuilder.java

```diff
@@ -43,9 +55,14 @@ private Future<Void> authorizedInternal(UserContext userContext, ResourceAction
                     )
                 );
             } else {
+                clock.stop(this.timer);
                 return Future.succeededFuture();
             }
-        }, t -> Future.failedFuture(new HttpException(HTTP_INTERNAL_ERROR, "failed to get user authorization")));
+        }, t -> {
+            clock.stop(this.timer);
+            return Future.failedFuture(new HttpException(HTTP_INTERNAL_ERROR, "failed to get user authorization"));
+        });
+
     }
```

Comment:
You might want to add a utility in FutureExtensions class,
that will allow you to do 
```java
return provider.isAuthorized(userContext, action, resourcePath).compose(authorized -> {
...
}).record(clock);
```

Reply:
move the future exetensions to common module